install-dev/upgrade/php/ps_1771_update_customer_note.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+/*
+ * Since note is now TYPE_STRING instead of TYPE_HTML it needs to be decoded
+ */
+function ps_1771_update_customer_note()
+{
+    $notes = Db::getInstance()->executeS(
+        'SELECT id_customer, note FROM ' . _DB_PREFIX_ . 'customer
+        WHERE note IS NOT NULL AND note != ""'
+    );
```

Comment:
my only concern here is the case where a shop has a lot of customers with notes, or even a lot of customers with notes on all of them. I the foreach below, it will make an update query foreach customer (potentially dozen of thousands on big shops...)

Not sure if having a batch mechanism here would be overkill or not (several updates per query)

Reply:
Since we don't support MyISAM we can't prepare SQL requests then execute like a batch.
Also, note values are different for each customer, let make requests easier to read, easier to manipulate, because during the upgrade we don't want to crash the system :sweat_smile:

Reply:
Well it's quite doable in php, but alright, it's probably overkill 👍   

I'm always a bit scared of update queries inside foreach when we don't know in advance the number of iterations... but I guess it's not a big deal